src/System.Web.HttpUtility/src/System/Web/HttpUtility.cs

```diff
@@ -0,0 +1,301 @@
+                throw new ArgumentNullException(nameof(query));
+
+            if (encoding == null)
+                throw new ArgumentNullException(nameof(encoding));
+
+            if ((query.Length > 0) && (query[0] == '?'))
+                query = query.Substring(1);
+
+            var result = new HttpQSCollection();
+            ParseQueryString(query, encoding, result);
+            return result;
+        }
+
+        private static void ParseQueryString(string query, Encoding encoding, NameValueCollection result)
+        {
+            if (query.Length == 0)
+                return;
+
+            var decoded = HtmlDecode(query);
+            var decodedLength = decoded.Length;
+            var namePos = 0;
+            var first = true;
+            while (namePos <= decodedLength)
+            {
+                int valuePos = -1, valueEnd = -1;
+                for (var q = namePos; q < decodedLength; q++)
+                    if ((valuePos == -1) && (decoded[q] == '='))
+                    {
+                        valuePos = q + 1;
+                    }
+                    else if (decoded[q] == '&')
+                    {
+                        valueEnd = q;
+                        break;
+                    }
+
+                if (first)
+                {
+                    first = false;
+                    if (decoded[namePos] == '?')
+                        namePos++;
+                }
+
+                string name;
+                if (valuePos == -1)
+                {
+                    name = null;
+                    valuePos = namePos;
+                }
+                else
+                {
+                    name = UrlDecode(decoded.Substring(namePos, valuePos - namePos - 1), encoding);
+                }
+                if (valueEnd < 0)
+                {
+                    namePos = -1;
+                    valueEnd = decoded.Length;
+                }
+                else
+                {
+                    namePos = valueEnd + 1;
+                }
+                var value = UrlDecode(decoded.Substring(valuePos, valueEnd - valuePos), encoding);
+
+                result.Add(name, value);
+                if (namePos == -1)
+                    break;
+            }
+        }
+
+        public static string HtmlDecode(string s)
+        {
+            return HttpEncoder.HtmlDecode(s);
+        }
+
+
+        public static void HtmlDecode(string s, TextWriter output)
+        {
+            HttpEncoder.HtmlDecode(s, output);
+        }
+
+
+        public static string HtmlEncode(string s)
+        {
+            return HttpEncoder.HtmlEncode(s);
+        }
+
+
+        public static string HtmlEncode(object value)
+        {
+            if (value == null)
+                return null;
+
+            return HtmlEncode(Convert.ToString(value, CultureInfo.CurrentCulture));
+        }
+
+
+        public static void HtmlEncode(string s, TextWriter output)
```

Comment:
We have an HtmlEncoder in System.Text.Web.Encodings:
https://github.com/dotnet/corefx/blob/master/src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs
Can we just call to that for the implementation?